### ISSUE TITLE
ngx-slider working with angular 16 ?

### ISSUE THREAD

**Issue by Urvishdudhagara** [angular-slider/ngx-slider#355]
I use ngx-slider in my project but facing error: 

Error: src/app/components/e-commerce/e-commerce.module.ts:67:5 - error NG6002: 'NgxSliderModule' does not appear to be an NgModule class.

67     NgxSliderModule
       ~~~~~~~~~~~~~~~

  node_modules/@angular-slider/ngx-slider/slider.module.d.ts:6:22
    6 export declare class NgxSliderModule {
                           ~~~~~~~~~~~~~~~
    This likely means that the library (@angular-slider/ngx-slider) which declares NgxSliderModule is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

**Comment by VicM** [comment]
Yesterday I face the same thing using angular 15.

Use the [ngx-slider-v2 package](https://www.npmjs.com/package/ngx-slider-v2?activeTab=readme)

Not sure that is the problem with the base one, but when installing the v2 version make sure to read the doc and install the proper package version. In your case:

npm install ngx-slider-v2@16

that will solve the problem, with that I am successfully using the slider.

**Comment by ccampanale** [comment]
I reviewed this "ngx-slider-v2" package and I'm hesitant to use it myself. The NPM package is rather misleading, linking back to this repository, documentation, etc. even though it is maintained in a different repository. Furthermore, the repository from which is it maintained appears to be a fork of a fork of this repository. I'm not explicitly saying this is an issue but it certainly raises eyebrows.

I personally would like to see official support for NG 14/15/16 from this repository.

**Comment by danisss9** [comment]
> I reviewed this "ngx-slider-v2" package and I'm hesitant to use it myself. The NPM package is rather misleading, linking back to this repository, documentation, etc. even though it is maintained in a different repository. Furthermore, the repository from which is it maintained appears to be a fork of a fork of this repository. I'm not explicitly saying this is an issue but it certainly raises eyebrows.
> 
> I personally would like to see official support for NG 14/15/16 from this repository.

I created the [ngx-slider-v2](https://www.npmjs.com/package/ngx-slider-v2?activeTab=readme) fork to update the lib to support Angular 14+ because the current oficial version does not support Ivy so it cannot be used in recent versions of Angular.

Altough it seems to be in the plans to update for Ivy support [see](https://github.com/angular-slider/ngx-slider/issues/351#issuecomment-1481086499), you can see my fork as a workaround until that isn't done.

**Comment by alex-matveev42** [comment]
@danisss9 when I try to use your slider with min = 25, max=68M and logScale = true browser is free and dead. Do you have any idea how I can avoid it?

**Comment by danisss9** [comment]
> @danisss9 when I try to use your slider with min = 25, max=68M, and logScale = true browser is freezed and then dead. Do you have any idea how I can avoid it?

There is a open issue on my Fork about a performance regression compared to this version, but i still didn't find the culprit, would appretiate help trying to debug it

**Comment by trckndrsch** [comment]
> @danisss9 when I try to use your slider with min = 25, max=68M, and logScale = true browser is freezed and then dead. Do you have any idea how I can avoid it?

I think there might be a workaround for your use case: https://github.com/danisss9/ngx-slider/issues/5#issuecomment-1684406881

**Comment by madmacc** [comment]
ngx-slider-v2@16.0.0 worked for me.
Waiting on implementation in this package.

**Comment by SebConejo** [comment]
It doesn't work with Angular 16. We need to wait for a release.
<img width="682" alt="Capture d’écran 2023-09-12 à 16 16 21" src="https://github.com/angular-slider/ngx-slider/assets/11723962/7f22b8bf-443d-4a73-9d05-bc70d1a5ea27">

**Comment by visuallization** [comment]
Making this package work with angular 16+ would be great! :)

**Comment by SebConejo** [comment]
@danisss9 Why not make a pull request of your work on v2? Maybe they'll be able to accept it quickly?

**Comment by danisss9** [comment]
> @danisss9 Why not make a pull request of your work on v2 on this repo ? Maybe they'll be able to accept it quickly?

I could do it, but I just updated the library itself, it still has some broken things like: the docs website generation scripts, some tests and github actions

**Comment by AlDrag** [comment]
@danisss9 let me know if you need a hand fixing that stuff, as I'm invested in keeping this project up for work.

**Comment by danisss9** [comment]
@AlDrag yes help would be apreciated and i dont mind making a pull request after all that is fixed

**Comment by piotrdz** [comment]
Hello everyone, original maintainer of ngx-slider here.

Long story short, this project has stagnated due to reasons better explained in Github issue #351.

But, there is good news. I am now in a position to return to this project and I am very much open to discussing its future, including taking contributions from pull requests and passing the admin duties to another trusted person.

I have created a Discord server called ngx-slider that I wish to use for discussion on the future of this project. Please join it using the link below:

https://discord.gg/C6K9S9r7

**Comment by Kaizodo** [comment]
Any update on this issue ?

**Comment by piotrdz** [comment]
@Kaizodo Yes, there is progress. @danisss9 has created a PR from his fork of the repository to upgrade to Angular 17 (#363) and I'm working on reviewing and merging it. I expect the v17 release could be ready by the end of next week (11th Feb).

**Comment by piotrdz** [comment]
A further update: I have merged the PR to support Angular 17 (#363) and other supporting work that I have done on top of that.

However, there has been a bit of a snag with converting end-to-end (e2e) tests from old protractor framework to new playwright framework. I would like to convert a few more of these tests to have at least a basic coverage of the most important test suites before making a final v17 release.

So the checklist for release as of now is to:
 - [ ] convert range slider e2e tests to playwright
 - [ ] convert vertical sliders e2e tests to playwright
 - [ ] do a few tests on mobile devices
 - [ ] test the packaged library when used as a dependency
 - [ ] make the release

I expect to tick off most of these items in the next day or two.

**Comment by piotrdz** [comment]
And we are done, I just published version 17.0.0 of the slider with Angular 17 support. Thank you for the patience.

Backports to previous Angular versions (14 through 16) will be completed in the near future.

I am closing this issue as the immediate concern of supporting new Angular versions has been addressed.

**Comment by piotrdz** [comment]
Work to add backports will be tracked in #368.
